src/utils/request-utils.ts

```diff
@@ -42,8 +42,45 @@ export const formDataToQuery = function (formdata: Record<string, any> | FormDat
 }
 
 export const getQueryParam = function (url: string, param: string): string {
+    // Check if URL and URLSearchParams are available in the global scope
+    // Fall back to manual parsing if URL/URLSearchParams aren't supported
+    if (typeof URL === 'undefined' || typeof URLSearchParams === 'undefined') {
+        return getQueryParamFallback(url, param)
+    }
+
+    // In case we fail to parse the URL, fall back to manual parsing
+    let urlObj: URL
+    try {
+        // Disabling compat/compat here because we've checked its existence above
+        urlObj = new URL(url) // eslint-disable-line compat/compat
+    } catch {
+        return getQueryParamFallback(url, param)
+    }
+
+    // Remove leading ? from search string if present
+    // This is not really necessary if there's a single ? in the search string
+    // but it's helpful if someone has some malformed URL with multiple ?s
+    const search = urlObj.search.replace(/^\?*/g, '')
+    const searchParams = new URLSearchParams(search) // eslint-disable-line compat/compat
+
+    let value = searchParams.get(param) ?? ''
+    try {
+        value = decodeURIComponent(value)
+    } catch {
+        logger.error('Skipping decoding for malformed query param: ' + value)
+    }
+
+    // Transforms + to space
+    return value.replace(/\+/g, ' ')
+}
+
+const getQueryParamFallback = function (url: string, param: string): string {
     const withoutHash: string = url.split('#')[0] || ''
-    const queryParams: string = withoutHash.split('?')[1] || ''
+
+    // By limiting the split to 2, we ensure that we only split on the first ?
+    // This is not really necessary if there's a single ? in the search string
+    // but it's helpful if someone has some malformed URL with multiple ?s
+    const queryParams: string = withoutHash.split('?', 2)[1] || ''
```

Comment:
Do we want to remove a potential leading `?` after this?

Reply:
Yeah, sounds good, let me add that too